cayenne-web/src/main/java/org/apache/cayenne/configuration/web/StatelessContextRequestHandler.java

```diff
@@ -67,6 +69,10 @@ public void requestStart(ServletRequest request, ServletResponse response) {
 
     public void requestEnd(ServletRequest request, ServletResponse response) {
         CayenneRuntime.bindThreadInjector(null);
+        ObjectContext context = BaseContext.getThreadObjectContext();
+        if (context != null) {
+        	((BaseContext)context).getQueryCache().clearLocalCache(Optional.empty());
```

Comment:
Don't like this `Optional.empty()` thing, it's a little bit obscure what is it doing and why it's always empty.
Maybe we can push it directly to  `NestedQueryCache` and call it explicitly from contexts that actually use it, but then we'll need some sort of `context.cleanup()` call.

Reply:
I don't like it either. The QueryCache interface doesn't have any concept of namespacing like the NestedQueryCache does, so any change that needs to target just the local cache and not shared&local must by necessity introduce something that represents the namespace or local/shared distinction. So I'm open to ideas. I'm not sure that NestedQueryCache really needs to exists as a separate class at all. The namespacing could be combined into to JCacheQueryCache as long as the QueryCache interface methods all support distinguishing between shared/local cache strategies, which it mostly does with this. The exception is removeGroup, which targets both shared and local, but that is probably fine as long as it is documented. I think combining NestedQueryCache would be my next choice.